### PR TITLE
Made Crafting GUIs tall

### LINKED ISSUE #2267: Crafting Status improvments
I think it would be really good if the crafting status window could be bigger because its difficult to actually see whats going on when you are requesting something more complex.
<a href="http://imgur.com/UkNAYD9.png"></a>
<a href="http://imgur.com/vhaqWua.png"></a>

I think it would be nice to have the option to also sort the crafting status with for example, crafting at the top, below that scheduled and at the bottom stored items.

### PR DESCRIPTION
**Pull request by thepaperpilot**
Partially resolves #2267

### CODE REVIEW

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 1]
```diff
@@ -576,5 +638,38 @@ protected void actionPerformed( final GuiButton btn ) throws IOException
...
+				final Enum cv = iBtn.getCurrentValue();
+				final Enum next = Platform.rotateEnum( cv, backwards, iBtn.getSetting().getPossibleValues() );
+
+				if( btn == this.terminalStyleBox )
+				{
+					AEConfig.instance().getConfigManager().putSetting( iBtn.getSetting(), next );
+				}
+				else
+				{
+					try
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: CRITICAL Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 2]
```diff
@@ -111,16 +128,45 @@ boolean isAutoStart()
...
+
 	@Override
 	public void initGui()
 	{
+		this.maxRows = this.getMaxRows();
+
+		final int magicNumber = ITEMSTACK_TOP_OFFSET + CPU_TOP_OFFSET;
+		final int extraSpace = this.height - magicNumber;
+
+		this.rows = (int) Math.floor( extraSpace / SECTION_HEIGHT );
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: MAJOR Remove this silly call to "Math.floor" [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2185)
MINOR Cast one of the operands of this division operation to a "double". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2184)

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 3]
```diff
@@ -60,7 +64,20 @@
 
 	private final ContainerCraftConfirm ccc;
 
-	private final int rows = 5;
+	private int rows = 5;
+	private int maxRows = Integer.MAX_VALUE;
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: MINOR Remove the "maxRows" field and declare it as a local variable in the relevant methods. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1450)

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 4]
```diff
@@ -390,17 +435,29 @@ public void drawFG( final int offsetX, final int offsetY, final int mouseX, fina
...
+		{
+			this.drawTexturedModalRect( offsetX, offsetY + 19 + x * SECTION_HEIGHT, 0, 19, x_width, SECTION_HEIGHT );
+		}
+
+		this.drawTexturedModalRect( offsetX, offsetY + 19 + this.rows * SECTION_HEIGHT - 1, 0, 133, x_width, 72 );
+	}
+
+	protected String getBackground()
+	{
+		return "guis/craftingreport.png";
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: MINOR Remove this method and declare a constant for this value. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3400)

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 5]
```diff
@@ -390,17 +435,29 @@ public void drawFG( final int offsetX, final int offsetY, final int mouseX, fina
 	@Override
 	public void drawBG( final int offsetX, final int offsetY, final int mouseX, final int mouseY )
 	{
-		this.setScrollBar();
-		this.bindTexture( "guis/craftingreport.png" );
-		this.drawTexturedModalRect( offsetX, offsetY, 0, 0, this.xSize, this.ySize );
+		this.bindTexture( this.getBackground() );
+		final int x_width = 238;
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: MINOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00117)

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 6]
```diff
@@ -576,5 +638,38 @@ protected void actionPerformed( final GuiButton btn ) throws IOException
...
+		{
+			final GuiImgButton iBtn = (GuiImgButton) btn;
+			if( iBtn.getSetting() != Settings.ACTIONS )
+			{
+				final Enum cv = iBtn.getCurrentValue();
+				final Enum next = Platform.rotateEnum( cv, backwards, iBtn.getSetting().getPossibleValues() );
+
+				if( btn == this.terminalStyleBox )
+				{
+					AEConfig.instance().getConfigManager().putSetting( iBtn.getSetting(), next );
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: MAJOR Method appeng.client.gui.implementations.GuiCraftConfirm.actionPerformed(GuiButton) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java [hunk 7]
```diff
@@ -132,23 +138,88 @@ protected void actionPerformed( final GuiButton btn ) throws IOException
...
 
 	@Override
 	public void initGui()
 	{
+		this.maxRows = this.getMaxRows();
+
+		final int magicNumber = ITEMSTACK_TOP_OFFSET + CANCEL_TOP_OFFSET - 1;
+		final int extraSpace = this.height - magicNumber;
+
+		this.rows = (int) Math.floor( extraSpace / SECTION_HEIGHT );
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java]: MAJOR Remove this silly call to "Math.floor" [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2185)
MINOR Cast one of the operands of this division operation to a "double". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2184)

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java [hunk 8]
```diff
@@ -341,8 +412,21 @@ public void drawFG( final int offsetX, final int offsetY, final int mouseX, fina
 	@Override
 	public void drawBG( final int offsetX, final int offsetY, final int mouseX, final int mouseY )
 	{
-		this.bindTexture( "guis/craftingcpu.png" );
-		this.drawTexturedModalRect( offsetX, offsetY, 0, 0, this.xSize, this.ySize );
+		this.bindTexture( this.getBackground() );
+		final int x_width = 238;
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java]: MINOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00117)

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java [hunk 9]
```diff
@@ -91,6 +94,9 @@
 	private List<IAEItemStack> visual = new ArrayList<IAEItemStack>();
 	private GuiButton cancel;
 	private int tooltip = -1;
+	private int rows = 6;
+	private int maxRows = Integer.MAX_VALUE;
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java]: MINOR Remove the "maxRows" field and declare it as a local variable in the relevant methods. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1450)

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java [hunk 10]
```diff
@@ -132,23 +138,88 @@ protected void actionPerformed( final GuiButton btn ) throws IOException
...
+
+			final GuiImgButton iBtn = (GuiImgButton) btn;
+			if( iBtn.getSetting() != Settings.ACTIONS )
+			{
+				final Enum cv = iBtn.getCurrentValue();
+				final Enum next = Platform.rotateEnum( cv, backwards, iBtn.getSetting().getPossibleValues() );
+
+				if( btn == this.terminalStyleBox )
+				{
+					AEConfig.instance().getConfigManager().putSetting( iBtn.getSetting(), next );
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java]: MAJOR Method appeng.client.gui.implementations.GuiCraftingCPU.actionPerformed(GuiButton) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java [hunk 11]
```diff
@@ -341,8 +412,21 @@ public void drawFG( final int offsetX, final int offsetY, final int mouseX, fina
...
+		{
+			this.drawTexturedModalRect( offsetX, offsetY + 19 + x * SECTION_HEIGHT, 0, 19, x_width, SECTION_HEIGHT );
+		}
+
+		this.drawTexturedModalRect( offsetX, offsetY + 19 + this.rows * SECTION_HEIGHT - 1, 0, 156, x_width, 27 );
+	}
+
+	protected String getBackground()
+	{
+		return "guis/craftingcpu.png";
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java]: MINOR Remove this method and declare a constant for this value. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3400)

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java [hunk 12]
```diff
@@ -132,23 +138,88 @@ protected void actionPerformed( final GuiButton btn ) throws IOException
...
+				final Enum cv = iBtn.getCurrentValue();
+				final Enum next = Platform.rotateEnum( cv, backwards, iBtn.getSetting().getPossibleValues() );
+
+				if( btn == this.terminalStyleBox )
+				{
+					AEConfig.instance().getConfigManager().putSetting( iBtn.getSetting(), next );
+				}
+				else
+				{
+					try
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java]: CRITICAL Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)

**Comment by orod-org** [comment]
SonarQube analysis reported 27 issues
* CRITICAL 3 critical
* MAJOR 15 major
* MINOR 9 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [GuiCraftConfirm.java#L607](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a42609871b0fff67f29f4db2dfa0fe8f2d41b762/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L607): The Cyclomatic Complexity of this method "actionPerformed" is 12 which is greater than 10 authorized. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. MAJOR [GuiCraftConfirm.java#L234](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a42609871b0fff67f29f4db2dfa0fe8f2d41b762/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L234): Method appeng.client.gui.implementations.GuiCraftConfirm.updateCPUButtonText() appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftConfirm.java#L337](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a42609871b0fff67f29f4db2dfa0fe8f2d41b762/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L337): Method appeng.client.gui.implementations.GuiCraftConfirm.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftConfirm.java#L362](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a42609871b0fff67f29f4db2dfa0fe8f2d41b762/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L362): Method appeng.client.gui.implementations.GuiCraftConfirm.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftConfirm.java#L387](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a42609871b0fff67f29f4db2dfa0fe8f2d41b762/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L387): Method appeng.client.gui.implementations.GuiCraftConfirm.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftingCPU.java#L341](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a42609871b0fff67f29f4db2dfa0fe8f2d41b762/src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java#L341): Method appeng.client.gui.implementations.GuiCraftingCPU.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftingCPU.java#L356](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a42609871b0fff67f29f4db2dfa0fe8f2d41b762/src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java#L356): Method appeng.client.gui.implementations.GuiCraftingCPU.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftingCPU.java#L371](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a42609871b0fff67f29f4db2dfa0fe8f2d41b762/src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java#L371): Method appeng.client.gui.implementations.GuiCraftingCPU.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiMEMonitorable.java#L77](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a42609871b0fff67f29f4db2dfa0fe8f2d41b762/src/main/java/appeng/client/gui/implementations/GuiMEMonitorable.java#L77): Unread field: appeng.client.gui.implementations.GuiMEMonitorable.offsetX; should this field be static? [rule](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ASS_SHOULD_BE_STATIC)
1. MAJOR [GuiMEMonitorable.java#L78](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/a42609871b0fff67f29f4db2dfa0fe8f2d41b762/src/main/java/appeng/client/gui/implementations/GuiMEMonitorable.java#L78): Unread field: appeng.client.gui.implementations.GuiMEMonitorable.lowerTextureOffset; should this field be static? [rule](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ASS_SHOULD_BE_STATIC)

**Comment by thepaperpilot** [comment]
Yikes, 27 issues? Most of those are how they already were elsewhere in the project.

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 13]
```diff
@@ -111,16 +128,45 @@ boolean isAutoStart()
...
+		}
+
 		super.initGui();
+		this.setScrollBar();
+
+		this.ySize = magicNumber + this.rows * SECTION_HEIGHT;
+		final int unusedSpace = this.height - this.ySize;
+		this.guiTop = (int) Math.floor( unusedSpace / ( unusedSpace < 0 ? 3.8f : 2.0f ) );
+		int offset = this.guiTop + 8;
+		this.setScrollBar();
```
Comment (yueh) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: Why  call `setScrollBar()` twice? (L157)

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java (hunk 3 above)
Comment (yueh) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: Having a `rows` and `maxRows` seems to be a bit redundant in their use. Moving it to a getter might also be an idea.

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 14]
```diff
@@ -111,16 +128,45 @@ boolean isAutoStart()
...
+		this.rows = (int) Math.floor( extraSpace / SECTION_HEIGHT );
+		if( this.rows > this.maxRows )
+		{
+			this.rows = this.maxRows;
+		}
+
+		if( this.rows < 5 )
+		{
+			this.rows = 5;
+		}
```
Comment (yueh) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: The whole sanitation should probably already happening in `getRows()` or `getMaxRows()`.

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 15]
```diff
@@ -447,6 +504,11 @@ public void postUpdate( final List<IAEItemStack> list, final byte ref )
 		this.setScrollBar();
 	}
 
+	int getMaxRows()
+	{
+		return AEConfig.instance().getConfigManager().getSetting( Settings.TERMINAL_STYLE ) == TerminalStyle.SMALL ? 5 : Integer.MAX_VALUE;
```
Comment (yueh) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: Should handle also the sanitation part. But for performance maybe keep `this.rows` and rename this to something like `calculateRows()` and call it during init?

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 16]
```diff
@@ -390,17 +439,24 @@ public void drawFG( final int offsetX, final int offsetY, final int mouseX, fina
 	@Override
 	public void drawBG( final int offsetX, final int offsetY, final int mouseX, final int mouseY )
 	{
-		this.setScrollBar();
 		this.bindTexture( "guis/craftingreport.png" );
-		this.drawTexturedModalRect( offsetX, offsetY, 0, 0, this.xSize, this.ySize );
+		final int x_width = 238;
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: MINOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00117)

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 17]
```diff
@@ -111,16 +127,32 @@ boolean isAutoStart()
...
+		this.buttonList.clear();
+		this.initGui();
+	}
+
 	@Override
 	public void initGui()
 	{
+		final int staticSpace = ITEMSTACK_TOP_OFFSET + CPU_TOP_OFFSET;
+
+		calculateRows(staticSpace);
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: MINOR '(' is not followed by whitespace. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
MINOR ')' is not preceded with whitespace. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)

File: src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java [hunk 18]
```diff
@@ -130,8 +162,29 @@ public void initGui()
...
+
+		this.terminalStyleBox = new GuiImgButton( this.guiLeft - 18, offset, Settings.TERMINAL_STYLE, AEConfig.instance().getConfigManager().getSetting( Settings.TERMINAL_STYLE ) );
+		this.buttonList.add( this.terminalStyleBox );
 	}
 
+	private void calculateRows( final int height )
+	{
+		final int maxRows = AEConfig.instance().getConfigManager().getSetting( Settings.TERMINAL_STYLE ) == TerminalStyle.SMALL ? 5 : Integer.MAX_VALUE;
+
+		final double extraSpace = this.height - height;
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java]: MINOR Cast one of the operands of this substraction operation to a "double". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2184)

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java [hunk 19]
```diff
@@ -132,23 +137,94 @@ protected void actionPerformed( final GuiButton btn ) throws IOException
...
+		this.terminalStyleBox = new GuiImgButton( this.guiLeft - 18, offset, Settings.TERMINAL_STYLE, AEConfig.instance().getConfigManager().getSetting( Settings.TERMINAL_STYLE ) );
 		this.buttonList.add( this.cancel );
+		this.buttonList.add( this.terminalStyleBox );
 	}
 
+	private void calculateRows( final int height )
+	{
+		final int maxRows = AEConfig.instance().getConfigManager().getSetting( Settings.TERMINAL_STYLE ) == TerminalStyle.SMALL ? 6 : Integer.MAX_VALUE;
+
+		final double extraSpace = this.height - height;
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java]: MINOR Cast one of the operands of this substraction operation to a "double". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2184)

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java [hunk 20]
```diff
@@ -342,7 +418,15 @@ public void drawFG( final int offsetX, final int offsetY, final int mouseX, fina
 	public void drawBG( final int offsetX, final int offsetY, final int mouseX, final int mouseY )
 	{
 		this.bindTexture( "guis/craftingcpu.png" );
-		this.drawTexturedModalRect( offsetX, offsetY, 0, 0, this.xSize, this.ySize );
+		final int x_width = 238;
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java]: MINOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00117)

File: src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java [hunk 21]
```diff
@@ -132,23 +137,94 @@ protected void actionPerformed( final GuiButton btn ) throws IOException
...
+		this.buttonList.clear();
+		this.initGui();
 	}
 
 	@Override
 	public void initGui()
 	{
+		final int staticSpace = ITEMSTACK_TOP_OFFSET + CANCEL_TOP_OFFSET - 1;
+
+		calculateRows(staticSpace);
```
Comment (orod-org) [inline src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java]: MINOR '(' is not followed by whitespace. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
MINOR ')' is not preceded with whitespace. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)

**Comment by orod-org** [comment]
SonarQube analysis reported 20 issues
* CRITICAL 3 critical
* MAJOR 9 major
* MINOR 8 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [GuiCraftConfirm.java#L601](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/caf9393225a4caaed83ef45237571434d99df9b6/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L601): The Cyclomatic Complexity of this method "actionPerformed" is 12 which is greater than 10 authorized. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. MAJOR [GuiCraftConfirm.java#L238](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/caf9393225a4caaed83ef45237571434d99df9b6/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L238): Method appeng.client.gui.implementations.GuiCraftConfirm.updateCPUButtonText() appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftConfirm.java#L341](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/caf9393225a4caaed83ef45237571434d99df9b6/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L341): Method appeng.client.gui.implementations.GuiCraftConfirm.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftConfirm.java#L366](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/caf9393225a4caaed83ef45237571434d99df9b6/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L366): Method appeng.client.gui.implementations.GuiCraftConfirm.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftConfirm.java#L391](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/caf9393225a4caaed83ef45237571434d99df9b6/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L391): Method appeng.client.gui.implementations.GuiCraftConfirm.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftingCPU.java#L346](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/caf9393225a4caaed83ef45237571434d99df9b6/src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java#L346): Method appeng.client.gui.implementations.GuiCraftingCPU.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftingCPU.java#L361](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/caf9393225a4caaed83ef45237571434d99df9b6/src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java#L361): Method appeng.client.gui.implementations.GuiCraftingCPU.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftingCPU.java#L376](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/caf9393225a4caaed83ef45237571434d99df9b6/src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java#L376): Method appeng.client.gui.implementations.GuiCraftingCPU.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

**Comment by orod-org** [comment]
SonarQube analysis reported 14 issues
* CRITICAL 3 critical
* MAJOR 9 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [GuiCraftConfirm.java#L601](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/20ea634e5e5459e2b6e89002674c662a7f892de1/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L601): The Cyclomatic Complexity of this method "actionPerformed" is 12 which is greater than 10 authorized. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. MAJOR [GuiCraftConfirm.java#L238](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/20ea634e5e5459e2b6e89002674c662a7f892de1/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L238): Method appeng.client.gui.implementations.GuiCraftConfirm.updateCPUButtonText() appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftConfirm.java#L341](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/20ea634e5e5459e2b6e89002674c662a7f892de1/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L341): Method appeng.client.gui.implementations.GuiCraftConfirm.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftConfirm.java#L366](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/20ea634e5e5459e2b6e89002674c662a7f892de1/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L366): Method appeng.client.gui.implementations.GuiCraftConfirm.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftConfirm.java#L391](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/20ea634e5e5459e2b6e89002674c662a7f892de1/src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java#L391): Method appeng.client.gui.implementations.GuiCraftConfirm.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftingCPU.java#L346](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/20ea634e5e5459e2b6e89002674c662a7f892de1/src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java#L346): Method appeng.client.gui.implementations.GuiCraftingCPU.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftingCPU.java#L361](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/20ea634e5e5459e2b6e89002674c662a7f892de1/src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java#L361): Method appeng.client.gui.implementations.GuiCraftingCPU.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [GuiCraftingCPU.java#L376](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/20ea634e5e5459e2b6e89002674c662a7f892de1/src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java#L376): Method appeng.client.gui.implementations.GuiCraftingCPU.drawFG(int, int, int, int) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

**Pull request closed without merging**